simulationworkflowschema/general.py

```diff
@@ -231,24 +285,261 @@ def normalize(self, archive, logger):
-                    input_structure = self.input_structure if n == 0 else self._calculations[n - 1].system_ref
+                    input_structure = (
+                        self.input_structure
+                        if n == 0
+                        else self._calculations[n - 1].system_ref
+                    )
                     if not input_structure:
                         input_structure = previous_structure
                     if input_structure:
-                        inputs.append(Link(name=INPUT_SYSTEM_NAME, section=input_structure))
+                        inputs.append(
+                            Link(name=INPUT_SYSTEM_NAME, section=input_structure)
+                        )
                     previous_structure = calculation.system_ref
-                    outputs.append(Link(name=OUTPUT_SYSTEM_NAME, section=calculation.system_ref))
+                    outputs.append(
+                        Link(name=OUTPUT_SYSTEM_NAME, section=calculation.system_ref)
+                    )
                 elif len(self._calculations) == len(self._systems):
-                    inputs.append(Link(name=INPUT_SYSTEM_NAME, section=self.input_structure if n == 0 else self._systems[n - 1]))
-                    outputs.append(Link(name=OUTPUT_SYSTEM_NAME, section=self._systems[n]))
+                    inputs.append(
+                        Link(
+                            name=INPUT_SYSTEM_NAME,
+                            section=self.input_structure
+                            if n == 0
+                            else self._systems[n - 1],
+                        )
+                    )
+                    outputs.append(
+                        Link(name=OUTPUT_SYSTEM_NAME, section=self._systems[n])
+                    )
                 else:
                     continue
                 if calculation.method_ref:
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=calculation.method_ref))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=calculation.method_ref)
+                    )
                 elif len(self._calculations) == len(self._methods):
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[n]))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[n])
+                    )
                 elif len(self._methods) == 1:
-                    inputs.append(Link(name=INPUT_METHOD_NAME, section=self._methods[0]))
-                self.tasks.append(Task(name=f'Step {n}', inputs=inputs, outputs=outputs))
+                    inputs.append(
+                        Link(name=INPUT_METHOD_NAME, section=self._methods[0])
+                    )
+                self.tasks.append(
+                    Task(name=f"Step {n}", inputs=inputs, outputs=outputs)
+                )
+
+
+class BeyondDFT2Tasks(SerialSimulation):
```

Comment:
should this be sepate file, beyond_dft.py perhaps.

Reply:
Ok, done.